pkg/apis/build/v1alpha1/trigger_types.go

```diff
@@ -0,0 +1,64 @@
+// Copyright The Shipwright Contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+package v1alpha1
+
+// WhenTypeName set of TriggerWhen valid names.
+type WhenTypeName string
+
+const (
+	// WhenTypeGitHub GitHub trigger type name.
+	WhenTypeGitHub WhenTypeName = "GitHub"
+
+	// WhenTypeImage Image trigger type name.
+	WhenTypeImage WhenTypeName = "Image"
+
+	// WhenTypePipeline Tekton Pipeline trigger type name.
+	WhenTypePipeline WhenTypeName = "Pipeline"
+)
+
+// GitHubEventName set of WhenGitHub valid event names.
+type GitHubEventName string
+
+const (
+	// GitHubPullRequestEvent github pull-request event name.
+	GitHubPullRequestEvent GitHubEventName = "PullRequest"
+
+	// GitHubPushEvent git push webhook event name.
+	GitHubPushEvent GitHubEventName = "Push"
+)
+
+// WhenImage attributes to match Image events.
+type WhenImage struct {
+	// Names fully qualified image names.
+	//
+	// +optional
+	Names []string `json:"names,omitempty"`
+}
+
+// WhenGitHub attributes to match GitHub events.
+type WhenGitHub struct {
+	// Events GitHub event names.
+	Events []GitHubEventName `json:"events,omitempty"`
```

Comment:
I guess this is not allowed to be empty.

```suggestion
	+kubebuilder:validation:MinItems=1
	Events []GitHubEventName `json:"events,omitempty"`
```